clock/mock.go

```diff
@@ -67,14 +89,17 @@ func (m *Mock) NewTimer(d time.Duration, tags ...string) *Timer {
 		nxt:  m.cur.Add(d),
 		mock: m,
 	}
-	m.addTimerLocked(t)
+	if d <= 0 {
+		// zero or negative duration timer means we should immediately fire
+		// it, rather than add it.
+		go t.fire(t.mock.cur)
+		return t
+	}
+	m.addEventLocked(t)
```

Comment:
nit: related to this PR, or did it sneak in from elsewhere? The behaviour does seem valid from what I can tell. https://go.dev/play/p/xYggZfm7Szw

Reply:
It shows up when downstack PRs get merged but this one hasn't yet been restacked